src/parser/SpatialQuery.cpp

```diff
@@ -133,30 +136,32 @@ SpatialQuery::SpatialQuery(const SparqlTriple& triple) {
                     "as predicate.");
   const std::string& input = triple.p_._iri;
 
-  if (input.starts_with(NEAREST_NEIGHBORS)) {
-    throw SpatialSearchException(
-        "The special predicate <nearest-neighbors:...> is no longer supported "
-        "due to confusing semantics. Please use SERVICE spatialSearch: {...} "
-        "instead. For information on its usage, see the QLever Wiki.");
-  }
-
   // Add variables to configuration object
   AD_CONTRACT_CHECK(triple.s_.isVariable() && triple.o_.isVariable(),
-                    "Currently, SpatialJoin needs two variables");
+                    "Currently, both the subject and the object of the triple "
+                    "that specifies a spatial join have to be variables.");
   setVariable("left", triple.s_, left_);
   setVariable("right", triple.o_, right_);
 
   // Helper to convert a ctre match to an integer
-  auto matchToInt = [](std::string_view match) {
-    AD_CORRECTNESS_CHECK(match.size() > 0);
+  auto matchToInt = [](std::string_view match) -> std::optional<size_t> {
+    if (match.size() == 0) {
+      return std::nullopt;
+    }
```

Comment:
Add a comment, that for the nearest neighbors search, the max dist is in fact optional and so the size can be 0 in this case. I just stumbled upon this.